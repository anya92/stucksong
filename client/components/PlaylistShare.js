import React from 'react';
import {
  FacebookShareButton,
  TwitterShareButton,
  WhatsappShareButton,
  TumblrShareButton,
  FacebookIcon,
  TwitterIcon,
  WhatsappIcon,
  TumblrIcon,
} from 'react-share';

const PlaylistShare = ({ url, name }) => {
  const iconOptions = {
    size: 50,
    logoFillColor: '#333',
    iconBgStyle: { fill: 'transparent' },
  };
  return (
    <div>
      <FacebookShareButton url={url} quote={name}>
        <FacebookIcon {...iconOptions} />
      </FacebookShareButton>
      <TwitterShareButton url={url} title={name}>
        <TwitterIcon {...iconOptions} />
      </TwitterShareButton>
      <WhatsappShareButton url={url} title={name}>
        <WhatsappIcon {...iconOptions} />
      </WhatsappShareButton>
      <TumblrShareButton url={url}>
        <TumblrIcon {...iconOptions} />
      </TumblrShareButton>
    </div>
  );
};

export default PlaylistShare;
