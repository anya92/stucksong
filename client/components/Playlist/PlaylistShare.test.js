import React from 'react';
import { shallow } from 'enzyme';

import PlaylistShare from './PlaylistShare';

describe('PlaylistShare component', () => {
  it('renders correctly', () => {
    const mockProps = {
      url: 'https://open.spotify.com/user/abc/playlist/123',
      name: 'Top 10 Tracks',
    };
    const wrapper = shallow(<PlaylistShare {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
