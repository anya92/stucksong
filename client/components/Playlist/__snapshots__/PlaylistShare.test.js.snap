// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistShare component renders correctly 1`] = `
<div>
  <CreatedButton
    quote="Top 10 Tracks"
    url="https://open.spotify.com/user/abc/playlist/123"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={
        Object {
          "fill": "transparent",
        }
      }
      logoFillColor="#333"
      size={50}
    />
  </CreatedButton>
  <CreatedButton
    title="Top 10 Tracks"
    url="https://open.spotify.com/user/abc/playlist/123"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={
        Object {
          "fill": "transparent",
        }
      }
      logoFillColor="#333"
      size={50}
    />
  </CreatedButton>
  <CreatedButton
    separator=" "
    title="Top 10 Tracks"
    url="https://open.spotify.com/user/abc/playlist/123"
    windowHeight={400}
    windowWidth={550}
  >
    <Icon
      className=""
      iconBgStyle={
        Object {
          "fill": "transparent",
        }
      }
      logoFillColor="#333"
      size={50}
    />
  </CreatedButton>
  <CreatedButton
    posttype="link"
    tags={Array []}
    url="https://open.spotify.com/user/abc/playlist/123"
    windowHeight={460}
    windowWidth={660}
  >
    <Icon
      className=""
      iconBgStyle={
        Object {
          "fill": "transparent",
        }
      }
      logoFillColor="#333"
      size={50}
    />
  </CreatedButton>
</div>
`;
